app/controllers/find/search/locations_controller.rb

```diff
@@ -5,6 +5,26 @@ def index
         @providers = RecruitmentCycle.current.providers.by_name_ascending
         @courses_by_location_or_training_provider_form = CoursesByLocationOrTrainingProviderForm.new
       end
+
+      def create
+        @courses_by_location_or_training_provider_form = CoursesByLocationOrTrainingProviderForm.new(params: find_courses_by_location_or_training_provider_form_params)
+
+        if @courses_by_location_or_training_provider_form.valid?
+          redirect_to find_age_groups_path
+        else
+          @providers = RecruitmentCycle.current.providers.by_name_ascending
```

Comment:
this and the same query on line 5 could use:

```ruby
def providers
  @providers ||= RecruitmentCycle.current.providers.by_name_ascending
end
```

Reply:
fixed